// node-onvif-ts/dist/node-onvif/module/service-base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnvifServiceBase = void 0;
var url_1 = require("url");
var soap_1 = require("./soap");
var OnvifServiceBase = /** @class */ (function () {
    function OnvifServiceBase(_a) {
        var xaddr = _a.xaddr, user = _a.user, pass = _a.pass;
        this.xaddr = '';
        this.user = '';
        this.pass = '';
        this.xaddr = xaddr;
        this.user = user || '';
        this.pass = pass || '';
        this.oxaddr = url_1.parse(this.xaddr);
        if (this.user) {
            this.oxaddr.auth = this.user + ':' + this.pass;
        }
        this.timeDiff = 0;
    }
    OnvifServiceBase.prototype.createRequestSoap = function (body) {
        return soap_1.createRequestSoap({
            body: body,
            xmlns: this.namespaceAttrList,
            diff: this.timeDiff,
            user: this.user,
            pass: this.pass,
        });
    };
    OnvifServiceBase.prototype.setAuth = function (user, pass) {
        this.user = user;
        this.pass = pass;
        if (this.user) {
            this.oxaddr.auth = this.user + ' ' + this.pass;
        }
        else {
            this.oxaddr.auth = '';
        }
    };
    return OnvifServiceBase;
}());
exports.OnvifServiceBase = OnvifServiceBase;
